### PR TITLE
Update Solidity files for HIP-358 (token create through the precompile)

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:
This PR updates the Solidity files `IHederaTokenService.sol` and `HederaTokenService.sol` with the most recent changes from HIP-358. It also adds new Solidity files that may serve as examples for token creation through the precompile contract:
* `KeyHelper.sol`, `FeeHelper.sol` and `ExpiryHelper.sol` contain helper methods 
*  `TokenCreateContract.sol` contains example methods for creating fungible and non-fungible tokens (both without and with fees) using the above mentioned helper contracts

### CODE REVIEW

**Review by stoqnkpL — APPROVED** [review]
LGTM!

File: hts-precompile/FeeHelper.sol [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+pragma experimental ABIEncoderV2;
+
+import "./HederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+import "./IHederaTokenService.sol";
+import "./KeyHelper.sol";
+
+contract FeeHelper is KeyHelper {
+
+    function createFixedFeeForHbars(
```
Comment (tinker-michaelj) [inline hts-precompile/FeeHelper.sol:12]: The name `createFixedHbarFee()` might be more readable.

File: hts-precompile/FeeHelper.sol [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        uint32 amount,
+        address feeCollector
+    ) internal pure returns (IHederaTokenService.FixedFee memory fixedFee) {
+        fixedFee.amount = amount;
+        fixedFee.useHbarsForPayment = true;
+        fixedFee.feeCollector = feeCollector;
+    }
+
+    function createFixedFeeForToken(
```
Comment (tinker-michaelj) [inline hts-precompile/FeeHelper.sol:21]: Similarly, `createFixedTokenFee()` here.

File: hts-precompile/FeeHelper.sol [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        uint32 amount,
+        address tokenId,
+        address feeCollector
+    ) internal pure returns (IHederaTokenService.FixedFee memory fixedFee) {
+        fixedFee.amount = amount;
+        fixedFee.tokenId = tokenId;
+        fixedFee.feeCollector = feeCollector;
+    }
+
+    function createFixedFeeForCurrentToken(
```
Comment (tinker-michaelj) [inline hts-precompile/FeeHelper.sol:31]: `createFixedSelfDenominatedFee()`

File: hts-precompile/FeeHelper.sol [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        uint32 amount,
+        address feeCollector
+    ) internal pure returns (IHederaTokenService.FixedFee memory fixedFee) {
+        fixedFee.amount = amount;
+        fixedFee.useCurrentTokenForPayment = true;
+        fixedFee.feeCollector = feeCollector;
+    }
+
+    function createFractionalFeeWithoutLimits(
```
Comment (tinker-michaelj) [inline hts-precompile/FeeHelper.sol:40]: `createFractionalFee()` and then below `createFractionalFeeWithMinAndMax()`, maybe?

File: hts-precompile/ExpiryHelper.sol [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.5.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "./HederaTokenService.sol";
+import "./FeeHelper.sol";
+
+contract ExpiryHelper is FeeHelper {
+
+    function getAutoRenewExpiry(
```
Comment (tinker-michaelj) [inline hts-precompile/ExpiryHelper.sol:10]: Maybe `createAutoRenewBasedExpiry()`?

**Review by tinker-michaelj — COMMENTED** [review]
Overall looks great, I made a few comments to maybe improve consistency of factory method names.

How much work is triggered by renaming methods?

**Pull request merged**